### ISSUE TITLE
Plots.jl pyplot backend segfaults in Pluto.jl

### ISSUE THREAD

**Issue by samuela** [fonsp/Pluto.jl#786]
Here's a reproduction notebook:
```julia
### A Pluto.jl notebook ###
# v0.12.17

using Markdown
using InteractiveUtils

# ╔═╡ 01447604-40b3-11eb-0da2-332083328ea2
import Plots

# ╔═╡ 07c1a6fc-40b3-11eb-136e-ad7229116481
Plots.pyplot()

# ╔═╡ 1a9bae18-40b3-11eb-2928-03bf53a21e2c
Plots.plot(1:10, rand(10))

# ╔═╡ Cell order:
# ╠═01447604-40b3-11eb-0da2-332083328ea2
# ╠═07c1a6fc-40b3-11eb-136e-ad7229116481
# ╠═1a9bae18-40b3-11eb-2928-03bf53a21e2c
```

When this is run I get
```
      From worker 6:
      From worker 6:	signal (11): Segmentation fault: 11
      From worker 6:	in expression starting at none:1
      From worker 6:	_ZNK7QScreen6handleEv at /Users/skainswo/.julia/conda/3/lib/libQt5Gui.5.9.7.dylib (unknown line)
      From worker 6:	Allocations: 84452654 (Pool: 84429752; Big: 22902); GC: 81Worker 6 terminated.
Distributed
.ProcessExitedException(6)
```
OTOH, running `julia --project pyplot.jl` works without issue.

Not only does the plot fail to show up, but I've found that it actually semi-crashes the Pluto system. Eg, when I go to rename the worksheet I get an error message:
<img width="353" alt="Screen Shot 2020-12-17 at 2 11 19 PM" src="https://user-images.githubusercontent.com/226872/102549896-c4abaf00-4071-11eb-8244-2c82b2033fc8.png">

along with some nasty output in the terminal:
```
.ProcessExitedException(6)Base.StackTraces.StackFrame[worker_from_id(::Distributed.ProcessGroup, ::Int64) at cluster.jl:1074, worker_from_id at cluster.jl:1071 [inlined], remotecall(::Function, ::Int64, ::Module, ::Vararg{Any,N} where N; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at remotecall.jl:376, remotecall at remotecall.jl:376 [inlined], macro expansion at task.jl:403 [inlined], macro expansion at macros.jl:224 [inlined], macro expansion at task.jl:332 [inlined], remotecall_eval(::Module, ::Array{Int64,1}, ::Expr) at macros.jl:218, eval_in_workspace(::Tuple{Pluto.ServerSession,Pluto.Notebook}, ::Expr) at WorkspaceManager.jl:251, cd_workspace(::Tuple{Pluto.ServerSession,Pluto.Notebook}, ::String) at WorkspaceManager.jl:83, move(::Pluto.ServerSession, ::Pluto.Notebook, ::String) at SessionActions.jl:76, (::Pluto.var"#177#179")(::Pluto.ServerSession, ::Dict{Any,Any}, ::Pluto.Notebook; initiator::Pluto.Initiator) at Dynamic.jl:140, (::Pluto.var"#177#179##kw")(::NamedTuple{(:initiator,),Tuple{Pluto.Initiator}}, ::Pluto.var"#177#179", ::Pluto.ServerSession, ::Dict{Any,Any}, ::Pluto.Notebook) at Dynamic.jl:138, process_ws_message(::Pluto.ServerSession, ::Dict{Any,Any}, ::HTTP.WebSockets.WebSocket{HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at WebServer.jl:354, (::Pluto.var"#215#225"{Pluto.ServerSession,Base.RefValue{Function}})(::HTTP.WebSockets.WebSocket{HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at WebServer.jl:160, upgrade(::Pluto.var"#215#225"{Pluto.ServerSession,Base.RefValue{Function}}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}; binary::Bool) at WebSockets.jl:160, upgrade at WebSockets.jl:142 [inlined], (::Pluto.var"#214#224"{Pluto.ServerSession,HTTP.Handlers.Router{Symbol("##269")},Base.RefValue{Function}})(::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at WebServer.jl:147, handle at Handlers.jl:270 [inlined], #4 at Handlers.jl:346 [inlined], macro expansion at Servers.jl:399 [inlined], (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.StreamHandlerFunction{Pluto.var"#214#224"{Pluto.ServerSession,HTTP.Handlers.Router{Symbol("##269")},Base.RefValue{Function}}}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at task.jl:356]
```

Just as a matter of clarification: what plotting libraries are supported in Pluto and which are not? It seems as though Plots.jl with the default gr backend works pretty well, and I've seen issues regarding other backend support but I'm not clear on what the status of support for each of these is.

**Comment by samuela** [comment]
Potentially related is https://github.com/fonsp/Pluto.jl/issues/511.

**Comment by fonsp** [comment]
Can you run this in the REPL:
```julia
pkg> activate --temp
pkg> add Plots PyPlot

julia> import Plots
julia> Plots.pyplot()
julia> p = Plots.plot(1:10, rand(10))

julia> Base.show(devnull, MIME"text/html"(), p)
julia> Base.show(devnull, MIME"image/svg"(), p)
julia> Base.show(devnull, MIME"image/png"(), p)
```

And share the output? (If it crashes then that isolates Pluto from this problem)

**Comment by fonsp** [comment]
It happens for me too. The problem occurs before the plot object is `Bae.show`n: if you add a semicolon then it also crashes.

**Comment by fonsp** [comment]
Manually starting a notebook process and running the code exactly like the Pluto server does not cause a crash:
```julia
julia> import Distributed

julia> Distributed.addprocs(1)
p = a1-element Array{Int64,1}:
 2

julia> p = ans
1-element Array{Int64,1}:
 2

julia> Distributed.remotecall_eval(Main, 2, quote
       include("/Users/fons/Documents/Pluto.jl/src/runner/Loader.jl")
       end)
3-element Array{String,1}:
 "@"
 "@v#.#"
 "@stdlib"

julia> Distributed.remotecall_eval(Main, 2, quote
       PlutoRunner.run_expression(quote
           import Plots
           Plots.pyplot()
           plot(1:10, rand(10))
       end, $(uuid1()), nothing
       end)
ERROR: syntax: missing comma or ) in argument list
Stacktrace:
 [1] top-level scope at none:1

julia> using UUIDs

julia> Distributed.remotecall_eval(Main, 2, quote
       PlutoRunner.run_expression(quote
           import Plots
           Plots.pyplot()
           plot(1:10, rand(10))
       end, $(uuid1()), nothing)
       end)
(nothing, 0x00000001598c1c6a)

julia> Distributed.remotecall_eval(Main, 2, :(1+1))
2

julia> # still alive
```

**Comment by fonsp** [comment]
Found it!

https://github.com/JuliaPy/PyPlot.jl/issues/512

**Comment by fonsp** [comment]
Fixed in 0.12.19!
